examples/versioning.ipynb

```diff
@@ -0,0 +1,667 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "07297a3d-048b-496b-adb0-8fdd67316021",
+   "metadata": {},
+   "source": [
+    "# Hosts: Working with asset Versions\n",
+    "\n",
+    "This notebook shows how to perform common versioning-related tasks using OpenAssetIO, the OpenAssetIO-MediaCreation traits, and a supporting manager.\n",
+    "\n",
+    "It uses a contrived library that contains several versions of the OpenAssetIO logo."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "60bd9de5-64c3-4f15-8824-ff1d94a894d7",
+   "metadata": {
+    "jp-MarkdownHeadingCollapsed": true
+   },
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a86e3e7c-6593-4bac-8e6b-c5a107dc702c",
+   "metadata": {
+    "jp-MarkdownHeadingCollapsed": true
+   },
+   "source": [
+    "#### Install packages"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "80285f6a-8134-4055-9430-a48a5f79e7d3",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Requirement already satisfied: openassetio in /Users/tomcowland/dev/OpenAssetIO-MediaCreation/.venv311/lib/python3.11/site-packages (1.0.0a14)\n",
+      "Requirement already satisfied: openassetio-manager-bal in /Users/tomcowland/dev/OpenAssetIO-MediaCreation/.venv311/lib/python3.11/site-packages (1.0.0a11)\n",
+      "Requirement already satisfied: openassetio-mediacreation in /Users/tomcowland/dev/OpenAssetIO-MediaCreation/.venv311/lib/python3.11/site-packages (1.0.0a7)\n",
+      "Requirement already satisfied: Pillow in /Users/tomcowland/dev/OpenAssetIO-MediaCreation/.venv311/lib/python3.11/site-packages (10.0.1)\n",
+      "Requirement already satisfied: importlib-metadata>=3.6 in /Users/tomcowland/dev/OpenAssetIO-MediaCreation/.venv311/lib/python3.11/site-packages (from openassetio) (6.8.0)\n",
+      "Requirement already satisfied: zipp>=0.5 in /Users/tomcowland/dev/OpenAssetIO-MediaCreation/.venv311/lib/python3.11/site-packages (from importlib-metadata>=3.6->openassetio) (3.17.0)\n",
+      "\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip available: \u001b[0m\u001b[31;49m22.3.1\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m23.2.1\u001b[0m\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpip install --upgrade pip\u001b[0m\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pip install openassetio openassetio-manager-bal openassetio-mediacreation Pillow"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6077b7b4-cbe7-40f2-908c-49c8e0a7d228",
+   "metadata": {
+    "jp-MarkdownHeadingCollapsed": true
+   },
+   "source": [
+    "#### Helpers"
```

Comment:
[Consider] I think the way to do this in reality would be to provide a single file python module in the jupyter directory itself, and do a local `pip install openassetio_jupyter_utilities/.`.

Having the utilities at the top of the file is good, but I think a bit jarring to tutorial flow, especailly as they're liable to get quite large.

Reply:
Good idea - separated into `resources/helpers.py`